staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -248,7 +255,14 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			}
 			metrics.WatchEvents.WithContext(req.Context()).WithLabelValues(kind.Group, kind.Version, kind.Kind).Inc()
 
-			if err := s.EmbeddedEncoder.Encode(event.Object, buf); err != nil {
+			if event.Type == watch.Bookmark && utilfeature.DefaultFeatureGate.Enabled(features.WatchList) {
+				shouldRecordWatchListLatency, err = storage.HasInitialEventsEndBookmarkAnnotation(event.Object)
```

Comment:
OK - so what you're doing is not even trying to understand whether user requests initial-list etc.
Instead you're assuming that lower layers will do the job and send (exactly one) initial-events-end bookmark correctly and just rely on that.

I guess I'm fine with that, but I would add a comment explaining that here.

Reply:
yes, it is a safe assumption, okay, I will add a comment.